### ISSUE TITLE
t/00-basic.t fails (NaN/Infinity sort order)

### ISSUE THREAD

**Issue by eserte** [rkleemann/Jasonify#3]
On some of my smokers the test suite fails:
```
# Failed test 'hash of numbers encodes correctly'
# at t/00-basic.t line 230.
# +---------------------------------+----+---------------------------------+
# | GOT                             | OP | CHECK                           |
# +---------------------------------+----+---------------------------------+
# | {"1234567890.123457" : 12345678 | eq | {"1234567890.123457" : 12345678 |
# | 90.123457, "1234567890.12346" : |    | 90.123457, "1234567890.12346" : |
# |  1234567890.12346, "9876543210" |    |  1234567890.12346, "9876543210" |
# |  : 9876543210, "NaN" : "NaN", " |    |  : 9876543210, "Infinity" : "In |
# | Infinity" : "Infinity"}         |    | finity", "NaN" : "NaN"}         |
# +---------------------------------+----+---------------------------------+

# Failed test 'hash of number references encodes correctly'
# at t/00-basic.t line 241.
# +---------------------------------+----+---------------------------------+
# | GOT                             | OP | CHECK                           |
# +---------------------------------+----+---------------------------------+
# | {"1234567890.12346" : true, "98 | eq | {"1234567890.12346" : true, "98 |
# | 76543210" : true, "NaN" : true, |    | 76543210" : true, "Infinity" :  |
# |  "Infinity" : true}             |    | true, "NaN" : true}             |
# +---------------------------------+----+---------------------------------+

# Failed test 'hash of hashes encodes correctly'
# at t/00-basic.t line 250.
# +---------------------------------+----+---------------------------------+
# | GOT                             | OP | CHECK                           |
# +---------------------------------+----+---------------------------------+
# | {"first" : {"0" : 0, "1" : 1, " | eq | {"first" : {"0" : 0, "1" : 1, " |
# | " : "", "false" : false, "true" |    | " : "", "false" : false, "true" |
# |  : true}, "second" : {"12345678 |    |  : true}, "second" : {"12345678 |
# | 90.123457" : 1234567890.123457, |    | 90.123457" : 1234567890.123457, |
# |  "1234567890.12346" : 123456789 |    |  "1234567890.12346" : 123456789 |
# | 0.12346, "9876543210" : 9876543 |    | 0.12346, "9876543210" : 9876543 |
# | 210, "NaN" : "NaN", "Infinity"  |    | 210, "Infinity" : "Infinity", " |
# | : "Infinity"}, "third" : {"1234 |    | NaN" : "NaN"}, "third" : {"1234 |
# | 567890.12346" : true, "98765432 |    | 567890.12346" : true, "98765432 |
# | 10" : true, "NaN" : true, "Infi |    | 10" : true, "Infinity" : true,  |
# | nity" : true}}                  |    | "NaN" : true}}                  |
# +---------------------------------+----+---------------------------------+
# Seeded srand with seed '20200229' from local date.
t/00-basic.t ........... 
Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/116 subtests 
```
It looks like expected and returned sort order of NaN and Infinity may differ.

**Comment by rkleemann** [comment]
That is really strange, because it should be using `cmp` to compare and sort those values, and `I` would come before `N` if that were the case.

Thanks for the report, I'm looking into it...

**Comment by rkleemann** [comment]
I've figured out what's going on here.  The external representation is `Infinity` or `NaN`, but the internal representation (and the way it's being sorted) is `inf` or `-nan`.  It seems to me that the correct solution is to translate the keys to their external representation before sorting them.  Doing that requires changes to [Datify](https://github.com/rkleemann/Datify), but I'll comment here when those changes have been implemented/released.
